### DISCUSSION TITLE
GPLv3 for ABoss sample code?

### DISCUSSION THREAD

**Discussion by bscottm** [aarondl/authboss#343] [General]
Of course, Authboss _really, completely and totally_ needs yet-another-sample-code repo. I'll admit that I fell into the Gin-Gonic trap, and I did get it working with the GORM SQLite driver for both user and session data. Needs a bit more work and documentation before I release it in the reasonably near future. The demo code is completely self-contained -- templates, user state and YAML configuration are all in one repo.

I'm considering releasing the code as Affero GPLv3. It's principally a defensive move to force anyone who derives from it to make their code GPL-ed.

Any concerns or objections of which I should be aware? Is a different license preferable?

**Comment by aarondl** [comment]
You're free to release your source code as you will. Authboss is MIT and that can be used as freely as you like. We all have differences of opinion when it comes to open source licensing, I think you should follow your heart on the subject. Just note that that will preclude it ever being turned into any official resource (if say it was to eventually supplant the current sample).

**Reply by bscottm** [reply]
The code is supposed to be a "HOWTO" -- from both our perspectives, no one should ever use Authboss example code as a code skeleton since no security audit has been done. GPLv3 doesn't preclude anyone from adapting the code; it should just make them think twice about using it verbatim.

That said, I am open to reconsidering licensing. I've released [authboss-worked](https://gitlab.com/scooter-phd/authboss-worked) on Gitlab. Feedback always welcomed.

Future thoughts: Adding Gorilla Mux and Chi web frameworks to build out a HOWTO.

(Disclaimer: I do/did have a relationship with the Software Freedom Law Center's founder and staff. Hence, my fondness for the GPL, after spending a week in NYC with SFLC and government IP attorneys and really getting to know the GPL.)
